code/modules/hydroponics/hydroponics.dm

```diff
@@ -903,10 +904,45 @@
 	weedlevel = Clamp(weedlevel + adjustamt, 0, 10)
 
 /obj/machinery/hydroponics/proc/spawnplant() // why would you put strange reagent in a hydro tray you monster I bet you also feed them blood
-	var/list/livingplants = list(/mob/living/simple_animal/hostile/tree, /mob/living/simple_animal/hostile/killertomato)
-	var/chosen = pick(livingplants)
-	var/mob/living/simple_animal/hostile/C = new chosen
-	C.faction = list("plants")
+	if(myseed)

```

Comment:
This should be a seperate pr

Reply:
what eve nis this for?

Reply:
I was found, that there is reaction with strange reagent supposed in hydroponics. But it was made incorrectly. At first I wanted to fix it. But after testing I desided, that it was kinda lame.So I improved it.

Reply:
@ninjanomnom if he fixed the bug I'd argue this counts as a refactor

Reply:
It's still quite out of scope for this pr and has nothing to do with the other changes